app/models/spree/product.rb

```diff
@@ -201,6 +201,17 @@ class Product < ApplicationRecord
 
     scope :active, lambda { where("spree_products.deleted_at IS NULL") }
 
+    scope :bulk_products, lambda { |user, params|
+                            ProductScopeQuery.new(user, params).bulk_products
+                          }
+
+    scope :paged_products_for_producers,
+          lambda { |user, params| ProductScopeQuery.new(user, params).paged_products_for_producers }
+
+    scope :product_scope, lambda { |user, params|
+                            ProductScopeQuery.new(user, params).product_scope
+                          }
+
```

Comment:
I know I said that scopes do generally belong into the model but it doesn't make sense to me here. These scopes are called from a single class and that class is already using ProductScopeQuery. The only benefit is visibility here, but guess what, the Spree::Product class is listed by Rubocop for having too many lines already.

Reply:
Yeah, I noticed that as well. We may have to trim it down eventually. I removed them and using the query object in the controller.